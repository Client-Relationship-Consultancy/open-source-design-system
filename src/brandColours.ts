export const colourPalette = {
  examplePalette: {
    primary: {
      main: {
        hex: "#A42824",
        RGB: "164,40,36",
        on: "#FFFFFF",
      },
      light: {
        hex: "#DA342F",
        RGB: "218,52,47",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#661614",
        RGB: "102,22,20",
        on: "#FFFFFF",
      },
    },
    secondary: {
      main: {
        hex: "#0B466B",
        RGB: "11,70,107",
        on: "#FFFFFF",
      },
      light: {
        hex: "#599CAF",
        RGB: "89,156,175",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#05263B",
        RGB: "5,38,59",
        on: "#FFFFFF",
      },
    },
    complimentary: {
      main: {
        hex: "#FFEE52",
        RGB: "255,238,82",
        on: "#272727",
      },
      light: {
        hex: "#FFF59D",
        RGB: "255,245,157",
        on: "#272727",
      },
      dark: {
        hex: "#CEB900",
        RGB: "206,185,0",
        on: "#272727",
      },
    },
    action: {
      main: {
        hex: "#743C8F",
        RGB: "116,60,143",
        on: "#FFFFFF",
      },
      light: {
        hex: "#AD65CF",
        RGB: "173,101,207",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#4E2662",
        RGB: "79,38,98",
        on: "#FFFFFF",
      },
    },
    error: {
      main: {
        hex: "#FF7320",
        RGB: "255,115,32",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#C54901",
        RGB: "197,73,1",
        on: "#FFFFFF",
      },
    },
    background: {
      hex: "#F6F6F6",
      RGB: "246,246,246",
    },
    surface: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    white: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    black: {
      main: { hex: "#272727", RGB: "39,39,39", on: "#FFFFFF" },
      tint80: { hex: "#696969", RGB: "105,105,105", on: "#FFFFFF" },
      tint60: { hex: "#A0A0A0", RGB: "160,160,160", on: "#FFFFFF" },
      tint40: { hex: "#C7C7C7", RGB: "199,199,199", on: "#272727" },
      tint20: { hex: "#ECECEC", RGB: "236,236,236", on: "#272727" },
    },
  },
};

interface IColourCodes {
  hex: string;
  RGB: string;
  on?: string;
}
interface IColourVariations {
  main: IColourCodes;
  light?: IColourCodes;
  dark?: IColourCodes;
}

interface IBlackVariations {
  main: IColourCodes;
  tint80: IColourCodes;
  tint60: IColourCodes;
  tint40: IColourCodes;
  tint20: IColourCodes;
}

export interface IColourPalette {
  primary: IColourVariations;
  secondary: IColourVariations;
  complimentary: IColourVariations;
  action: IColourVariations;
  error: IColourVariations;
  background: IColourCodes;
  surface: IColourCodes;
  white: IColourCodes;
  black: IBlackVariations;
}
