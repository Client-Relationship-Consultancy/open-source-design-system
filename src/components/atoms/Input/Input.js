import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import { colourPalette } from "../../../brandColours"

// Decalre CSS for StyledInput with styled component
const StyledInput = styled.input`
  display: block;
  box-sizing: border-box;
  outline: none;
  border: none;
  border-bottom: 1px solid ${props => props.theme.black.tint40.hex};
  padding: 0.5rem;
  font-size: 1rem;
  font-family: "Gentona", "Montserrat";
  transition: 0.3s ease all;
  color: ${props => props.theme.black.main.hex};
  :active,
  :focus {
    border-bottom: 2px solid ${props => props.theme.primary.main.hex};
    + label {
      color: ${props => props.theme.primary.main.hex};
    }
  }
`
StyledInput.defaultProps = {
  theme: colourPalette.examplePalette,
}
StyledInput.displayName = "StyledInput"

const StyledTextArea = styled.textarea`
  display: block;
  box-sizing: border-box;
  outline: none;
  border: 1px solid ${props => props.theme.black.tint40.hex};
  padding: 0.5rem;
  font-size: 1rem;
  font-family: "Gentona", "Montserrat";
  transition: 0.3s ease all;
  color: ${props => props.theme.black.main.hex};
  :active,
  :focus {
    border-bottom: 2px solid ${props => props.theme.primary.main.hex};
    + label {
      color: ${props => props.theme.primary.main.hex};
    }
  }
  border-radius: 5px;
  margin-top: 0.5rem;
  height: 10rem;
  border-bottom: 1px solid ${props => props.theme.black.tint40.hex};
`
StyledTextArea.defaultProps = {
  theme: colourPalette.examplePalette,
}
StyledTextArea.displayName = "StyledInput"

// Declare React Component Input
const Input = props => {
  // Refactored values out of props
  const {
    className,
    type,
    onChange,
    placeholder,
    id,
    name,
    value,
    required,
    disabled,
    multiLine = false,
  } = props

  const InputType = multiLine ? StyledTextArea : StyledInput
  const inputClassName = className ? `Input ${className}` : "Input"
  return (
    <InputType
      className={inputClassName}
      type={type}
      onChange={onChange}
      placeholder={placeholder}
      id={id}
      name={name}
      value={value}
      required={required}
      disabled={disabled}
    />
  )
}

// Declare the display name of component for static build
// Display name should match export name
Input.displayName = "Input"
// Declare propTypes of Input
Input.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  disabled: PropTypes.bool,
  /** Placeholder is not a replacement for label as it is not optimal for handicapped users */
  placeholder: PropTypes.string,
  /** Needed for controlled inputs */
  onChange: PropTypes.func,
  /** Needed for controlled inputs */
  value: PropTypes.string,
  required: PropTypes.bool,
  className: PropTypes.string,
  multiLine: PropTypes.bool,
}

export default Input
