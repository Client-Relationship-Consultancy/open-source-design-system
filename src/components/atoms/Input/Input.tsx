import React from "react";
import styled from "styled-components";

import { colourPalette } from "../../../brandColours";

const Container = styled.div``;

const StyledInput = styled.input`
  display: block;
  box-sizing: border-box;
  outline: none;
  border: none;
  border-bottom: 1px solid ${props => props.theme.black.tint40.hex};
  padding: 0.5rem;
  font-size: 1rem;
  font-family: "Gentona", "Montserrat";
  transition: 0.3s ease all;
  color: ${props => props.theme.black.main.hex};
  :active,
  :focus {
    border-bottom: 2px solid ${props => props.theme.primary.main.hex};
    + label {
      color: ${props => props.theme.primary.main.hex};
    }
  }
`;

StyledInput.defaultProps = {
  theme: colourPalette.examplePalette,
};

StyledInput.displayName = "StyledInput";


interface IProps extends React.HTMLProps<HTMLInputElement> {
  multiLine: boolean;
  icon: React.ReactNode;
}

const Input: React.FC<IProps> = (props: IProps ) => {
  const {
    className,
    type,
    onChange,
    placeholder,
    id,
    name,
    value,
    required,
    disabled,
    icon,
  } = props;

  const inputClassName = className ? `Input ${className}` : "Input";
  return (
    <Container>
      {icon || null}
      <StyledInput
        className={inputClassName}
        type={type}
        onChange={onChange}
        placeholder={placeholder}
        id={id}
        name={name}
        value={value}
        required={required}
        disabled={disabled}
      />
    </Container>
  );
};

export default Input;
