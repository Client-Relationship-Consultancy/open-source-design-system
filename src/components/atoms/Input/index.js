import Input, { IInputProps } from "./Input"

export { IInputProps }
export default Input
