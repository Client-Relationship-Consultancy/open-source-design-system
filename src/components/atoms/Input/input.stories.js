import React from "react"
import { storiesOf } from "@storybook/react"
import Input from "./Input"

const stories = storiesOf("Input", module)

stories.add("Standard style", context => {
  const text = {
    description: [
      "The Card compoennt style is simple and clear to help bring its contents forward. Cards are the top level container of most components.",
      "This component doesn't have to have a header or footer.",
    ],
    dos: [
      "Use this component as a top level wrapper",
      "Only use the footer for placing buttons and interactive elements",
      "Buttons can be placed in the body as well as the footer",
      "Do make sure the content are all aligned",
      "Do use the box shadow option when the Card is on a white background",
    ],
    donts: [
      "Do not nest Card components within each other",
      "Do not use the box shadow option when on a coloured background",
    ],
  }

  Object.assign(context, text)

  return (
  <div>
    <Input />
  </div>
  )
})
