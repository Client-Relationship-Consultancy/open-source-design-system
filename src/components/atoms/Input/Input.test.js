import React from "react"
import { mount } from "enzyme"
import "jest-styled-components"
import Input, { Container, StyledInput } from "./Input"
import { colourPalette } from "../../../brandColours"

const { black } = colourPalette.examplePalette

describe("Input", () => {
  const commonProps = {
    value: "Hello",
    name: "example",
    type: "text",
    onChange: () => {},
  }

  it("render input with value", () => {
    const component = mount(<Input {...commonProps} />)
    expect(component.prop("value")).toEqual("Hello")
  })

  it("render matches last snapshot", () => {
    const component = mount(<Input {...commonProps} />)
    expect(component).toMatchSnapshot()
  })

  it("render matches last snapshot with classname", () => {
    const customProps = {
      ...commonProps,
      className: "testClassName",
    }
    const component = mount(<Input {...customProps} />)
    expect(component.props().className).toEqual("testClassName")
  })
})

describe("<Container />", () => {
  it("should match last snapshot", () => {
    const component = mount(<Container />)
    expect(component).toMatchSnapshot()
  })

  it("should have background colour as transparent if no bgColor is passed", () => {
    const component = mount(<Container />)
    expect(component).toHaveStyleRule("background-color", "transparent")
  })

  it("should have background colour if bgColor is passed", () => {
    const component = mount(<Container bgColor="red" />)
    expect(component).toHaveStyleRule("background-color", "red")
  })

  it("should have no border if border is not passed", () => {
    const component = mount(<Container />)
    expect(component).toHaveStyleRule("none")
  })

  it("should have bottom border if border is passed bottom", () => {
    const component = mount(<Container border="bottom" />)
    expect(component).toHaveStyleRule("border", undefined)
    expect(component).toHaveStyleRule("border-bottom", `1px solid ${black.tint60.hex}`)
  })

  it("should have a border if border is passed all", () => {
    const component = mount(<Container border="all" />)
    expect(component).toHaveStyleRule("border", `1px solid ${black.tint60.hex}`)
  })
})

describe("<StyledInput />", () => {
  it("should match last snapshot", () => {
    const component = mount(<StyledInput />)
    expect(component).toMatchSnapshot()
  })
  it("should take the color if passed in as prop", () => {
    const component = mount(<StyledInput color="red" />)
    expect(component).toHaveStyleRule("color", "red")
  })
})
