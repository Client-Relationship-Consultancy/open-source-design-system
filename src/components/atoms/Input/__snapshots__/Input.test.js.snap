// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render matches last snapshot 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
  outline: none;
  border: none;
  border-bottom: 1px solid #C7C7C7;
  padding: 0.5rem;
  font-size: 1rem;
  font-family: "Gentona","Montserrat";
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
  color: #272727;
}

.c0:active,
.c0:focus {
  border-bottom: 2px solid #A42824;
}

.c0:active + label,
.c0:focus + label {
  color: #A42824;
}

<Input
  name="example"
  onChange={[Function]}
  type="text"
  value="Hello"
>
  <StyledInput
    name="example"
    onChange={[Function]}
    theme={
      Object {
        "action": Object {
          "dark": Object {
            "RGB": "79,38,98",
            "hex": "#4E2662",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "173,101,207",
            "hex": "#AD65CF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "116,60,143",
            "hex": "#743C8F",
            "on": "#FFFFFF",
          },
        },
        "background": Object {
          "RGB": "246,246,246",
          "hex": "#F6F6F6",
        },
        "black": Object {
          "main": Object {
            "RGB": "39,39,39",
            "hex": "#272727",
            "on": "#FFFFFF",
          },
          "tint20": Object {
            "RGB": "236,236,236",
            "hex": "#ECECEC",
            "on": "#272727",
          },
          "tint40": Object {
            "RGB": "199,199,199",
            "hex": "#C7C7C7",
            "on": "#272727",
          },
          "tint60": Object {
            "RGB": "160,160,160",
            "hex": "#A0A0A0",
            "on": "#FFFFFF",
          },
          "tint80": Object {
            "RGB": "105,105,105",
            "hex": "#696969",
            "on": "#FFFFFF",
          },
        },
        "complimentary": Object {
          "dark": Object {
            "RGB": "206,185,0",
            "hex": "#CEB900",
            "on": "#272727",
          },
          "light": Object {
            "RGB": "255,245,157",
            "hex": "#FFF59D",
            "on": "#272727",
          },
          "main": Object {
            "RGB": "255,238,82",
            "hex": "#FFEE52",
            "on": "#272727",
          },
        },
        "error": Object {
          "dark": Object {
            "RGB": "197,73,1",
            "hex": "#C54901",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "255,115,32",
            "hex": "#FF7320",
            "on": "#FFFFFF",
          },
        },
        "primary": Object {
          "dark": Object {
            "RGB": "102,22,20",
            "hex": "#661614",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "218,52,47",
            "hex": "#DA342F",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "164,40,36",
            "hex": "#A42824",
            "on": "#FFFFFF",
          },
        },
        "secondary": Object {
          "dark": Object {
            "RGB": "5,38,59",
            "hex": "#05263B",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "89,156,175",
            "hex": "#599CAF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "11,70,107",
            "hex": "#0B466B",
            "on": "#FFFFFF",
          },
        },
        "surface": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
        "white": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
      }
    }
    type="text"
    value="Hello"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "_foldedDefaultProps": Object {
            "theme": Object {
              "action": Object {
                "dark": Object {
                  "RGB": "79,38,98",
                  "hex": "#4E2662",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "173,101,207",
                  "hex": "#AD65CF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "116,60,143",
                  "hex": "#743C8F",
                  "on": "#FFFFFF",
                },
              },
              "background": Object {
                "RGB": "246,246,246",
                "hex": "#F6F6F6",
              },
              "black": Object {
                "main": Object {
                  "RGB": "39,39,39",
                  "hex": "#272727",
                  "on": "#FFFFFF",
                },
                "tint20": Object {
                  "RGB": "236,236,236",
                  "hex": "#ECECEC",
                  "on": "#272727",
                },
                "tint40": Object {
                  "RGB": "199,199,199",
                  "hex": "#C7C7C7",
                  "on": "#272727",
                },
                "tint60": Object {
                  "RGB": "160,160,160",
                  "hex": "#A0A0A0",
                  "on": "#FFFFFF",
                },
                "tint80": Object {
                  "RGB": "105,105,105",
                  "hex": "#696969",
                  "on": "#FFFFFF",
                },
              },
              "complimentary": Object {
                "dark": Object {
                  "RGB": "206,185,0",
                  "hex": "#CEB900",
                  "on": "#272727",
                },
                "light": Object {
                  "RGB": "255,245,157",
                  "hex": "#FFF59D",
                  "on": "#272727",
                },
                "main": Object {
                  "RGB": "255,238,82",
                  "hex": "#FFEE52",
                  "on": "#272727",
                },
              },
              "error": Object {
                "dark": Object {
                  "RGB": "197,73,1",
                  "hex": "#C54901",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "255,115,32",
                  "hex": "#FF7320",
                  "on": "#FFFFFF",
                },
              },
              "primary": Object {
                "dark": Object {
                  "RGB": "102,22,20",
                  "hex": "#661614",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "218,52,47",
                  "hex": "#DA342F",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "164,40,36",
                  "hex": "#A42824",
                  "on": "#FFFFFF",
                },
              },
              "secondary": Object {
                "dark": Object {
                  "RGB": "5,38,59",
                  "hex": "#05263B",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "89,156,175",
                  "hex": "#599CAF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "11,70,107",
                  "hex": "#0B466B",
                  "on": "#FFFFFF",
                },
              },
              "surface": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
              "white": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
            },
          },
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Input__StyledInput-sc-27uu00-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "display:block;box-sizing:border-box;outline:none;border:none;border-bottom:1px solid ",
              [Function],
              ";padding:0.5rem;font-size:1rem;font-family:\\"Gentona\\",\\"Montserrat\\";transition:0.3s ease all;color:",
              [Function],
              ";:active,:focus{border-bottom:2px solid ",
              [Function],
              ";+ label{color:",
              [Function],
              ";}}",
            ],
          },
          "displayName": "StyledInput",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Input__StyledInput-sc-27uu00-0",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      name="example"
      onChange={[Function]}
      theme={
        Object {
          "action": Object {
            "dark": Object {
              "RGB": "79,38,98",
              "hex": "#4E2662",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "173,101,207",
              "hex": "#AD65CF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "116,60,143",
              "hex": "#743C8F",
              "on": "#FFFFFF",
            },
          },
          "background": Object {
            "RGB": "246,246,246",
            "hex": "#F6F6F6",
          },
          "black": Object {
            "main": Object {
              "RGB": "39,39,39",
              "hex": "#272727",
              "on": "#FFFFFF",
            },
            "tint20": Object {
              "RGB": "236,236,236",
              "hex": "#ECECEC",
              "on": "#272727",
            },
            "tint40": Object {
              "RGB": "199,199,199",
              "hex": "#C7C7C7",
              "on": "#272727",
            },
            "tint60": Object {
              "RGB": "160,160,160",
              "hex": "#A0A0A0",
              "on": "#FFFFFF",
            },
            "tint80": Object {
              "RGB": "105,105,105",
              "hex": "#696969",
              "on": "#FFFFFF",
            },
          },
          "complimentary": Object {
            "dark": Object {
              "RGB": "206,185,0",
              "hex": "#CEB900",
              "on": "#272727",
            },
            "light": Object {
              "RGB": "255,245,157",
              "hex": "#FFF59D",
              "on": "#272727",
            },
            "main": Object {
              "RGB": "255,238,82",
              "hex": "#FFEE52",
              "on": "#272727",
            },
          },
          "error": Object {
            "dark": Object {
              "RGB": "197,73,1",
              "hex": "#C54901",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "255,115,32",
              "hex": "#FF7320",
              "on": "#FFFFFF",
            },
          },
          "primary": Object {
            "dark": Object {
              "RGB": "102,22,20",
              "hex": "#661614",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "218,52,47",
              "hex": "#DA342F",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "164,40,36",
              "hex": "#A42824",
              "on": "#FFFFFF",
            },
          },
          "secondary": Object {
            "dark": Object {
              "RGB": "5,38,59",
              "hex": "#05263B",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "89,156,175",
              "hex": "#599CAF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "11,70,107",
              "hex": "#0B466B",
              "on": "#FFFFFF",
            },
          },
          "surface": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
          "white": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
        }
      }
      type="text"
      value="Hello"
    >
      <input
        className="c0"
        name="example"
        onChange={[Function]}
        type="text"
        value="Hello"
      />
    </StyledComponent>
  </StyledInput>
</Input>
`;

exports[`Input render matches last snapshot with multiline 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
  outline: none;
  border: 1px solid #C7C7C7;
  padding: 0.5rem;
  font-size: 1rem;
  font-family: "Gentona","Montserrat";
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
  color: #272727;
  border-radius: 5px;
  margin-top: 0.5rem;
  height: 10rem;
  border-bottom: 1px solid #C7C7C7;
}

.c0:active,
.c0:focus {
  border-bottom: 2px solid #A42824;
}

.c0:active + label,
.c0:focus + label {
  color: #A42824;
}

<Input
  multiLine={true}
  name="example"
  onChange={[Function]}
  type="text"
  value="Hello"
>
  <StyledInput
    name="example"
    onChange={[Function]}
    theme={
      Object {
        "action": Object {
          "dark": Object {
            "RGB": "79,38,98",
            "hex": "#4E2662",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "173,101,207",
            "hex": "#AD65CF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "116,60,143",
            "hex": "#743C8F",
            "on": "#FFFFFF",
          },
        },
        "background": Object {
          "RGB": "246,246,246",
          "hex": "#F6F6F6",
        },
        "black": Object {
          "main": Object {
            "RGB": "39,39,39",
            "hex": "#272727",
            "on": "#FFFFFF",
          },
          "tint20": Object {
            "RGB": "236,236,236",
            "hex": "#ECECEC",
            "on": "#272727",
          },
          "tint40": Object {
            "RGB": "199,199,199",
            "hex": "#C7C7C7",
            "on": "#272727",
          },
          "tint60": Object {
            "RGB": "160,160,160",
            "hex": "#A0A0A0",
            "on": "#FFFFFF",
          },
          "tint80": Object {
            "RGB": "105,105,105",
            "hex": "#696969",
            "on": "#FFFFFF",
          },
        },
        "complimentary": Object {
          "dark": Object {
            "RGB": "206,185,0",
            "hex": "#CEB900",
            "on": "#272727",
          },
          "light": Object {
            "RGB": "255,245,157",
            "hex": "#FFF59D",
            "on": "#272727",
          },
          "main": Object {
            "RGB": "255,238,82",
            "hex": "#FFEE52",
            "on": "#272727",
          },
        },
        "error": Object {
          "dark": Object {
            "RGB": "197,73,1",
            "hex": "#C54901",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "255,115,32",
            "hex": "#FF7320",
            "on": "#FFFFFF",
          },
        },
        "primary": Object {
          "dark": Object {
            "RGB": "102,22,20",
            "hex": "#661614",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "218,52,47",
            "hex": "#DA342F",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "164,40,36",
            "hex": "#A42824",
            "on": "#FFFFFF",
          },
        },
        "secondary": Object {
          "dark": Object {
            "RGB": "5,38,59",
            "hex": "#05263B",
            "on": "#FFFFFF",
          },
          "light": Object {
            "RGB": "89,156,175",
            "hex": "#599CAF",
            "on": "#FFFFFF",
          },
          "main": Object {
            "RGB": "11,70,107",
            "hex": "#0B466B",
            "on": "#FFFFFF",
          },
        },
        "surface": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
        "white": Object {
          "RGB": "255,255,255",
          "hex": "#FFFFFF",
          "on": "#272727",
        },
      }
    }
    type="text"
    value="Hello"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "_foldedDefaultProps": Object {
            "theme": Object {
              "action": Object {
                "dark": Object {
                  "RGB": "79,38,98",
                  "hex": "#4E2662",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "173,101,207",
                  "hex": "#AD65CF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "116,60,143",
                  "hex": "#743C8F",
                  "on": "#FFFFFF",
                },
              },
              "background": Object {
                "RGB": "246,246,246",
                "hex": "#F6F6F6",
              },
              "black": Object {
                "main": Object {
                  "RGB": "39,39,39",
                  "hex": "#272727",
                  "on": "#FFFFFF",
                },
                "tint20": Object {
                  "RGB": "236,236,236",
                  "hex": "#ECECEC",
                  "on": "#272727",
                },
                "tint40": Object {
                  "RGB": "199,199,199",
                  "hex": "#C7C7C7",
                  "on": "#272727",
                },
                "tint60": Object {
                  "RGB": "160,160,160",
                  "hex": "#A0A0A0",
                  "on": "#FFFFFF",
                },
                "tint80": Object {
                  "RGB": "105,105,105",
                  "hex": "#696969",
                  "on": "#FFFFFF",
                },
              },
              "complimentary": Object {
                "dark": Object {
                  "RGB": "206,185,0",
                  "hex": "#CEB900",
                  "on": "#272727",
                },
                "light": Object {
                  "RGB": "255,245,157",
                  "hex": "#FFF59D",
                  "on": "#272727",
                },
                "main": Object {
                  "RGB": "255,238,82",
                  "hex": "#FFEE52",
                  "on": "#272727",
                },
              },
              "error": Object {
                "dark": Object {
                  "RGB": "197,73,1",
                  "hex": "#C54901",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "255,115,32",
                  "hex": "#FF7320",
                  "on": "#FFFFFF",
                },
              },
              "primary": Object {
                "dark": Object {
                  "RGB": "102,22,20",
                  "hex": "#661614",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "218,52,47",
                  "hex": "#DA342F",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "164,40,36",
                  "hex": "#A42824",
                  "on": "#FFFFFF",
                },
              },
              "secondary": Object {
                "dark": Object {
                  "RGB": "5,38,59",
                  "hex": "#05263B",
                  "on": "#FFFFFF",
                },
                "light": Object {
                  "RGB": "89,156,175",
                  "hex": "#599CAF",
                  "on": "#FFFFFF",
                },
                "main": Object {
                  "RGB": "11,70,107",
                  "hex": "#0B466B",
                  "on": "#FFFFFF",
                },
              },
              "surface": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
              "white": Object {
                "RGB": "255,255,255",
                "hex": "#FFFFFF",
                "on": "#272727",
              },
            },
          },
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Input__StyledTextArea-sc-27uu00-1",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "display:block;box-sizing:border-box;outline:none;border:1px solid ",
              [Function],
              ";padding:0.5rem;font-size:1rem;font-family:\\"Gentona\\",\\"Montserrat\\";transition:0.3s ease all;color:",
              [Function],
              ";:active,:focus{border-bottom:2px solid ",
              [Function],
              ";+ label{color:",
              [Function],
              ";}}border-radius:5px;margin-top:0.5rem;height:10rem;border-bottom:1px solid ",
              [Function],
              ";",
            ],
          },
          "displayName": "StyledInput",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Input__StyledTextArea-sc-27uu00-1",
          "target": "textarea",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      name="example"
      onChange={[Function]}
      theme={
        Object {
          "action": Object {
            "dark": Object {
              "RGB": "79,38,98",
              "hex": "#4E2662",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "173,101,207",
              "hex": "#AD65CF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "116,60,143",
              "hex": "#743C8F",
              "on": "#FFFFFF",
            },
          },
          "background": Object {
            "RGB": "246,246,246",
            "hex": "#F6F6F6",
          },
          "black": Object {
            "main": Object {
              "RGB": "39,39,39",
              "hex": "#272727",
              "on": "#FFFFFF",
            },
            "tint20": Object {
              "RGB": "236,236,236",
              "hex": "#ECECEC",
              "on": "#272727",
            },
            "tint40": Object {
              "RGB": "199,199,199",
              "hex": "#C7C7C7",
              "on": "#272727",
            },
            "tint60": Object {
              "RGB": "160,160,160",
              "hex": "#A0A0A0",
              "on": "#FFFFFF",
            },
            "tint80": Object {
              "RGB": "105,105,105",
              "hex": "#696969",
              "on": "#FFFFFF",
            },
          },
          "complimentary": Object {
            "dark": Object {
              "RGB": "206,185,0",
              "hex": "#CEB900",
              "on": "#272727",
            },
            "light": Object {
              "RGB": "255,245,157",
              "hex": "#FFF59D",
              "on": "#272727",
            },
            "main": Object {
              "RGB": "255,238,82",
              "hex": "#FFEE52",
              "on": "#272727",
            },
          },
          "error": Object {
            "dark": Object {
              "RGB": "197,73,1",
              "hex": "#C54901",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "255,115,32",
              "hex": "#FF7320",
              "on": "#FFFFFF",
            },
          },
          "primary": Object {
            "dark": Object {
              "RGB": "102,22,20",
              "hex": "#661614",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "218,52,47",
              "hex": "#DA342F",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "164,40,36",
              "hex": "#A42824",
              "on": "#FFFFFF",
            },
          },
          "secondary": Object {
            "dark": Object {
              "RGB": "5,38,59",
              "hex": "#05263B",
              "on": "#FFFFFF",
            },
            "light": Object {
              "RGB": "89,156,175",
              "hex": "#599CAF",
              "on": "#FFFFFF",
            },
            "main": Object {
              "RGB": "11,70,107",
              "hex": "#0B466B",
              "on": "#FFFFFF",
            },
          },
          "surface": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
          "white": Object {
            "RGB": "255,255,255",
            "hex": "#FFFFFF",
            "on": "#272727",
          },
        }
      }
      type="text"
      value="Hello"
    >
      <textarea
        className="c0"
        name="example"
        onChange={[Function]}
        type="text"
        value="Hello"
      />
    </StyledComponent>
  </StyledInput>
</Input>
`;
